src/Io/KeyboardActionMapping.h

```diff
@@ -29,22 +29,31 @@ namespace Io {
 
         void MapKey(InputAction action, PlatformKey key);
         void MapKey(InputAction action, PlatformKey key, KeyToggleType type);
+        void MapGamepadKey(InputAction action, PlatformKey key);
         PlatformKey MapDefaultKey(InputAction action);
 
         PlatformKey GetKey(InputAction action) const;
+        PlatformKey GetGamepadKey(InputAction action) const;
         KeyToggleType GetToggleType(InputAction action) const;
+        bool CompareKey(InputAction action, PlatformKey key) const;
```

Comment:
You can't do that as a single key could correspond to multiple InputActions.

Reply:
That's... a very questionable design.

Are we talking about a single key doing different things in different contexts, or users purposefully binding several input actions to the same key?

Reply:
It was mentioned in PR description:
`controls key change menu doesn't like the same keys for multiple actions and we have those now like C for spellbook and character tab or R for rest menu and rest for 8 hours.`

So yeah, same keys are used for different things in different menus, they were just hardcoded previously so wasn't really very apparent. If you remember @pskelton even mentioned that invoking rest menu and actual resting action in this menu is the same key. But there is no really another way without key duplicates, especially if you want to support things like gamepads which are having very limited set of available buttons. For example I binded rest menu by default on "back" key, and rest/wait actions on usual A/B/X/Y which are doing different things when you are actually in-game and not some menu... Some things I think could be removed, e.g. I'm pretty sure nobody ever used Yes/No hotkeys and used Escape/Trigger instead of them...

Reply:
Yeah, makes sense. One way to fix this would be to introduce an additional enum of contexts/screens, but I guess we'd rather not go this way.

OK, let's leave it the way it is. I'd only suggest renaming the method to clearly convey the meaning. Something like IsKeyForAction.

Reply:
We even have this enum, it is `WindowType`. But IMHO we should really decide what we will do with our GUI code first... If we throw it out and rewrite it/replace it with some ready solution then there is no point to invest time in these things as they most likely will be thrown out along with it. It is just temporary solution to have somewhat working hotkeys for timebeing until we come up with a plan to deal with our GUI. And investing further time in it only make sense if we will decide to spend enormous amount of time to try refactor it to state that working with it not becoming nightmare every single time...

Reply:
Understanding our long-term plans for GUI is important, but this shouldn't stop us from making sure that the code we already have is sound.

As I said, I suggest we just rename this function for now, to better convey the meaning of what it's doing.

Regarding the GUI plans, let's discuss it in Discord.